CMakeLists.txt

```diff
@@ -15,12 +15,32 @@ endif()
 
 project(half)
 find_package(ROCM REQUIRED)
+
 include(ROCMSetupVersion)
 include(ROCMCreatePackage)
 
+# FOR HANDLING ENABLE/DISABLE OPTIONAL BACKWARD COMPATIBILITY for FILE/FOLDER REORG
+option(BUILD_FILE_REORG_BACKWARD_COMPATIBILITY "Build with file/folder reorg with backward compatibility enabled" ON)
+
 rocm_setup_version(VERSION 1.12.0 NO_GIT_TAG_VERSION)
 
-install(FILES include/half.hpp DESTINATION include)
+install(FILES include/half.hpp DESTINATION include/half)
+
+if(BUILD_FILE_REORG_BACKWARD_COMPATIBILITY)
+#Generate header template file
+  file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/wrapper/header.hpp.in "/* \nCopyright (c) 2022 Advanced Micro Devices, Inc. All rights reserved.\n*/
+    \n\n#ifndef @include_guard@\n#define @include_guard@
+    \n\n#pragma message(\"This file is deprecated. Use file from include path /opt/rocm-ver/include and prefix with half\")
+    \n@include_statements@ \n\n#endif")
```

Comment:
I believe rocm-cmake has wrappers to do this automatically.

Reply:
Yes, Since it was only one file, was trying to independently here to lighten the changes. Will add rocm-cmake wrapper as suggested.